pkg/kn/commands/service/export.go

```diff
@@ -181,20 +224,49 @@ func exportServiceWithActiveRevisions(latestSvc *servingv1.Service, client clien
 	return exportedSvcList, nil
 }
 
-func setTrafficSplit(latestSvc *servingv1.Service, exportedSvc servingv1.Service) servingv1.Service {
+func exportActiveRevisionList(latestSvc *servingv1.Service, client clientservingv1.KnServingClient) (*servingv1.Service, *servingv1.RevisionList, error) {
+	var exportedRevItems []servingv1.Revision
 
-	exportedSvc.Spec.RouteSpec = latestSvc.Spec.RouteSpec
-	return exportedSvc
+	//get revisions to export from traffic
+	revsMap := getRevisionstoExport(latestSvc)
+
+	// Query for list with filters
+	revisionList, err := client.ListRevisions(clientservingv1.WithService(latestSvc.ObjectMeta.Name))
```

Comment:
This code (up to `sortRevisions`) is shared with the other method so it could be extracted into an own function.